zarr/_storage/store.py

```diff
@@ -129,6 +130,34 @@ def _ensure_store(store: Any):
             f"wrap it in Zarr.storage.KVStore. Got {store}"
         )
 
+    def getitems(
+        self, keys: Sequence[str], contexts: Mapping[str, Context] = {}
+    ) -> Mapping[str, Any]:
+        """Retrieve data from multiple keys.
+
+        Parameters
+        ----------
+        keys : Iterable[str]
+            The keys to retrieve
+        contexts: Mapping[str, Context]
+            A mapping of keys to their context. Each context is a mapping of store
+            specific information. E.g. a context could be a dict telling the store
+            the preferred output array type: `{"meta_array": cupy.empty(())}`
```

Comment:
I don't understand why the context needs to be key specific. If a user wants to load data into the GPU, surely they want to do that for the entire read, not just some chunks. It seems like we could simplify this a lot by just providing one single context to `getitems`, rather than a mapping.

There is also likely a performance cost to doing it this way, no? If you're reading 1000s of chunks at a time, it seems silly to be carrying around all these big mappings to the same context.

Reply:
Agree, this is to support the generic _context_ approach. Although, it is possible to limit the performance impact by using a custom `Mapping` that returns the same value for all keys.

Also notice, the current code implements the context approach where the `meta_array` argument is embedded within the context argument. We could rollback to my original approach where `getitems()` takes the `meta_array` directly (`getitems(..., meta_array:ndarray`)